src/rng.rs

```diff
@@ -392,10 +396,36 @@ macro_rules! impl_fill {
 impl_fill!(u16, u32, u64, usize, u128,);
 impl_fill!(i8, i16, i32, i64, isize, i128,);
 
+macro_rules! impl_fill_delegate_to_maybe_uninit{
+    () => {};
+    ($t:ty) => {
+        impl Fill for [$t] {
+            fn try_fill<R: Rng + ?Sized>(&mut self, rng: &mut R) -> Result<(), Error> {
+                Fill::try_fill(
+                    unsafe {
+                        slice::from_raw_parts_mut(self.as_mut_ptr()
+                            as *mut mem::MaybeUninit<$t>,
+                            self.len()
+                        )
+                    },
+                    rng,
+                )
+            }
+        }
+    };
+    ($t:ty, $($tt:ty,)*) => {
+        impl_fill_delegate_to_maybe_uninit!($t);
+        impl_fill_delegate_to_maybe_uninit!($($tt,)*);
+    };
+}
+
+impl_fill_delegate_to_maybe_uninit!(bool, char, f32, f64,);
```

Comment:
*This* part doesn't make much sense to me: replacing what was a safe impl of `Fill for [bool]` (etc.) with an `unsafe` one which lowers to `[MaybeUninit<bool>]` (etc.).